rustls/src/msgs/message.rs

```diff
@@ -66,7 +68,7 @@ impl MessagePayload {
 pub struct OpaqueMessage {
     pub typ: ContentType,
     pub version: ProtocolVersion,
-    pub payload: Payload,
+    pub payload: Vec<u8>,
```

Comment:
Isn't this supposed to be `PayloadU16` because we need to be able to encode the length in 16 bits. See the unchecked use of `as` in the implementation here. It seems a bit sad to lose that info in the type. OTOH, `PayloadU16::new` doesn't check that the input's length is small enough to be encodable either.

Ultimately it seems like we should be making the `Payload*` types contain a `Cow` or `T: AsRef<[u8]>` or similar.

Anyway, I don't have super strong feelings about this but I'm definitely missing context to offer good feedback.

Reply:
I've removed this commit for now.